Holtron.Net/Network/Encryption/NetEncryptionAESGCM.cs

```diff
@@ -0,0 +1,82 @@
+using System.Security.Cryptography;
+
+namespace Holtron.Net.Network.Encryption
+{
+    public class NetEncryptionAESGCM : INetEncryption
+    {
+        // This implementation is from
+        // https://stackoverflow.com/questions/60889345/using-the-aesgcm-class
+        private readonly AesGcm aesGcm;
+
+        public NetEncryptionAESGCM(byte[] key)
+        {
+            var derivedKey = new Rfc2898DeriveBytes(key, new byte[8], 1000).GetBytes(16);
+            aesGcm = new AesGcm(derivedKey);
+        }
+
+        public NetEncryptionAESGCM(string key)
+        {
+            var derivedKey = new Rfc2898DeriveBytes(key, new byte[8], 1000).GetBytes(16);
+            aesGcm = new AesGcm(derivedKey);
+        }
+
+        public NetEncryptionAESGCM(Span<byte> key)
+        {
+            var derivedKey = new Rfc2898DeriveBytes(key.ToArray(), new byte[8], 1000).GetBytes(16);
+            aesGcm = new AesGcm(derivedKey);
+        }
+
+        public bool Decrypt(NetIncomingMessage message)
+        {
+            var unencryptedLengthBits = (int)message.ReadUInt32();
+            var encryptedDataLength = message.ReadInt32();
+            var encryptedData = message.ReadBytes(encryptedDataLength).AsSpan();
+            var nonce = encryptedData.Slice(0, 12);
+            var tag = encryptedData.Slice(12, 12);
+            var cipherText = encryptedData.Slice(24, encryptedData.Length - 24);
+            var plainTextMessage = new byte[cipherText.Length];
+
+            aesGcm.Decrypt(nonce, cipherText, tag, plainTextMessage);
+
+            message.Data = plainTextMessage.ToArray();
```

Comment:
It looks like setting `message.Data` sets `message.m_data` under the hood. You probably only need one or the other.

Reply:
Standardized using `message.m_data`.